apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableClientPartial.cs

```diff
@@ -1017,13 +1074,25 @@ internal BigtableClientImpl(BigtableServiceApiClient client, string appProfileId
             _appProfileId = appProfileId;
         }
 
+        // TODO: Add a public constructor after multi-channel support?
+        internal BigtableClientImpl(string appProfileId, params BigtableServiceApiClient[] clients)
+        {
+            _clients = clients;
+            _appProfileId = appProfileId;
+        }
+
         // TODO: Add Multi-channel support
         /// <inheritdoc/>
-        private BigtableServiceApiClient GetUnderlyingClient() => _client;
+        private BigtableServiceApiClient GetUnderlyingClient() => _clients.Length > 1 ? _clients[GetClientIndex()] : _clients[0];
+
+        private int GetClientIndex()
+        {
+            return (Interlocked.Increment(ref _clientNumber) - 1) % _clients.Length;
```

Comment:
I would initialize `_clientNumber` to `-1` and then remove the subtraction (and outer parens) here.